sys/shell/commands/sc_ipv6_nc.c

```diff
@@ -168,14 +168,46 @@ static int _ipv6_nc_del(char *ipv6_addr_str)
     return 0;
 }
 
+static int _ipv6_nc_reset(void)
+{
+    ng_ipv6_nc_t *tmp = NULL;
+
+    for (ng_ipv6_nc_t *entry = ng_ipv6_nc_get_next(NULL);
+         entry != NULL;
+         tmp = entry, entry = ng_ipv6_nc_get_next(entry)) {
+        if (tmp) {
+            ng_ipv6_nc_remove(tmp->iface, &tmp->ipv6_addr);
+        }
+    }
+
+    /* remove last entry */
+    if (tmp) {
+        ng_ipv6_nc_remove(tmp->iface, &tmp->ipv6_addr);
+    }
+
+    printf("success: reset neighbor cache\n");
+
+    return 0;
```

Comment:
I thought about using the init function. But then I saw that the remove
function is also doing some cleanup like releasing pktsnips and alike. Is
it safe to just fire init?
Am 06.08.2015 00:05 schrieb "Martine Lenders" <notifications@github.com>:

> In sys/shell/commands/sc_ipv6_nc.c
> <https://github.com/RIOT-OS/RIOT/pull/3572#discussion_r36360310>:
>
> > +    for (ng_ipv6_nc_t *entry = ng_ipv6_nc_get_next(NULL);
> > +         entry != NULL;
> > +         tmp = entry, entry = ng_ipv6_nc_get_next(entry)) {
> > +        if (tmp) {
> > +            ng_ipv6_nc_remove(tmp->iface, &tmp->ipv6_addr);
> > +        }
> > +    }
> > +
> > +    /* remove last entry */
> > +    if (tmp) {
> > +        ng_ipv6_nc_remove(tmp->iface, &tmp->ipv6_addr);
> > +    }
> > +
> > +    printf("success: reset neighbor cache\n");
> > +
> > +    return 0;
>
> You can just use ng_ipv6_nc_init() here ;-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RIOT-OS/RIOT/pull/3572/files#r36360310>.
>